src/components/image-select.tsx

```diff
@@ -56,18 +53,21 @@ class ImageSelect extends React.Component<GridModalProps, GridModalState> {
       return;
     }
 
-    if (!this.validMessage(data)) {
+    const gridImage: CropSelection | undefined = GridValidators.iframePostMessage(data)
+
+    if(!gridImage) {
       return;
     }
 
-    const imageUrl = event.data.crop.data.master.secureUrl;
+    const imageUrl: URL = gridImage?.crop.data.master?.secureUrl!
```

Comment:
The property of master is set to optional but you have marked this as not being undefined. How can you be sure of this and is it worth checking?

Reply:
The model comes from the [case class](https://github.com/guardian/grid/blob/5ca61af6ecaecffc4acc142511bd596dd698b693/common-lib/src/main/scala/com/gu/mediaservice/model/Crop.scala#L11).

`master` was introduced post Grid launching so had to be an `Option` for backwards compatibility. It would be interesting to see how many crops do not have a master and remove the `Option`. I'd suspect there would only be a handful.

If `master` is undefined here, it would result in a runtime error. I don't think we'd get into that situation as we'll be using fresh crops.

Reply:
You could add to the if statement on line 58 to avoid the runtime issue.

Reply:
I think its best to solve this within the grid-client - providing a function that get's the highest quality image, returning the master when present or the largest asset.

I'm slowly adding this to grid-client; I'll close this PR for now to reduce noise.